Automattic/wp-calypso
# Billing: Update ChangePaymentMethod to use PaymentMethodSelector

#### Changes proposed in this Pull Request

This abstracts the payment method list and assignment code from `ChangePaymentMethod` into its own component called `PaymentMethodSelector`. The primary purpose of this is to enable reusing that component in other payment method forms like `AddNewPaymentMethod`.

Part of https://github.com/Automattic/wp-calypso/issues/48966

Depends on https://github.com/Automattic/wp-calypso/pull/49057

<img width="948" alt="Screen Shot 2021-01-19 at 7 54 56 PM" src="https://user-images.githubusercontent.com/2036909/105112117-50a96000-5a90-11eb-8391-455516f715eb.png">

<img width="1008" alt="Screen Shot 2021-01-19 at 7 55 26 PM" src="https://user-images.githubusercontent.com/2036909/105112124-53a45080-5a90-11eb-8990-5be36e699b3f.png">

#### Testing instructions

- Visit the subscription page for a purchase that already has a payment method assigned.
- Click on the "Change payment method" button.
- Verify that the list of payment methods appears.
- Select the "new" credit card payment method, fill it in, and submit the form.
- Verify that you are redirected to the subscription page with a success message.
- Verify that the new payment method is created and assigned to the subscription.
- Repeat the above steps but select an existing card and verify that everything works as before.
